service/header/interface.go

```diff
@@ -5,14 +5,25 @@ import (
 	"errors"
 	"fmt"
 
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	tmbytes "github.com/tendermint/tendermint/libs/bytes"
 )
 
+// Validator aliases a func that validates ExtendedHeader.
+type Validator = func(context.Context, *ExtendedHeader) pubsub.ValidationResult
```

Comment:
Remember when we couldn't do this 😂

Reply:
Not really. We might have some misunderstanding. Good that we are on the same page in the end. 

One thing I would change here is to remove dependency on pubsub type, but this is fine for now to keep as otherwise that would overengineering with no real upsides.

Reply:
resolved.